File: backend/ShiftMate.API/Models/ShiftAssignment.cs
namespace ShiftMate.API.Models;

public class ShiftAssignment
{
    public int Id { get; set; }

    public int ShiftId { get; set; }

    public int EmployeeId { get; set; }

    public DateTime AssignedAt { get; set; }

    public string Status { get; set; } = string.Empty;

    public Shift Shift { get; set; } = null!;

    public Employee Employee { get; set; } = null!;
}

File: frontend/src/types/shiftAssignment.ts
export interface ShiftAssignment {
    id: number
    shiftId: number
    employeeId: number
    employeeName: string
    locationName: string
    shiftStartTime: string
    shiftEndTime: string
    assignedAt: string
    status: string
}

export  interface AssignEmployeeToShiftRequest {
    shiftId: number
    employeeId: number
}